src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.skia.cs

```diff
@@ -228,6 +229,7 @@ private void UpdateTextBoxView()
 						_caretRect.Height = Math.Ceiling(rect.Height);
 						_caretRect.SetValue(Canvas.LeftProperty, rect.Left);
 						_caretRect.SetValue(Canvas.TopProperty, rect.Top);
+						_caretRect.Fill = DefaultBrushes.TextForegroundBrush;
```

Comment:
Is this called in case the theme is changed too?

Reply:
Yes. I've tested it and it immediately worked, so I didn't think too much about it. Theoretically, changing the theme would cause a redraw (and maybe an entire layout cycle?) that would lead to CaretFound firing. I should probably invalidate the inlines just to make sure. Nice catch.